src/pytorch_lightning/strategies/fully_sharded_native.py

```diff
@@ -117,7 +121,7 @@ def __init__(
         mixed_precision: Optional[MixedPrecision] = None,
         **kwargs: Any,
     ) -> None:
-        if not _TORCH_GREATER_EQUAL_1_12:
+        if not _TORCH_GREATER_EQUAL_1_12 or not torch.distributed.is_available():
```

Comment:
Why add this? What's the stacktrace if you remove it?

Reply:
Removing it `import pytorch_lightning` reports no error. But it could make sense to raise the subsequent MisconfigurationException when torch>=1.12.0 but distributed is not available.

Reply:
When would this class get instantiated with distributed unavailable?

If we want to add this check, it would need a different error message

Reply:
honestly, I have no idea. I'll remove the check!